application/models/TemplateConfiguration.php

```diff
@@ -1354,4 +1380,188 @@ public function setOptionKeysToInherit(){
+        if (!empty($prefix)) {
+            $filteredCategories = array_filter($categoryList, function ($v) use ($prefix) { return $v->pathPrefix == $prefix; });
+            if (empty($filteredCategories)) {
+                return null;    // No category matched the path's prefix
+            }
+            $category = reset($filteredCategories);
+
+            // Validate that the file exists
+            $realPath = realpath($category->path . '/' . substr($path, strlen($prefix)));
+
+            // If the file exists and no traversing is done (the real path starts with the category's base path),
+            // return the real path
+            if ($realPath !== false && strpos($realPath, $category->path) === 0) {
+                $virtualPath = str_replace($category->path, $category->pathPrefix, $realPath);
+                return new ThemeFileInfo($realPath, $virtualPath, $category);
+            } else {
+                return null;
+            }
+        }
+
+        // Path doesn't match the virtual path category format, so we try the determine if it belongs to a category.
+        foreach ($categoryList as $category) {
+            // Check for paths relative to the theme, relative to the rootdir, or absolute
+            $realPath = realpath($category->path . '/' . $path);
+            if ($realPath === false) {
+                $realPath = realpath(Yii::app()->getConfig('rootdir') . '/' . $path);
+            }
+            if ($realPath === false && strpos($path, '/') === 0) {
+                $realPath = realpath($path);
+            }
+            if ($realPath === false) {
+                continue;
+            }
+
+            // If the real path is within a category's path, we return it.
+            if (strpos($realPath, $category->path) === 0) {
+                $virtualPath = str_replace($category->path, $category->pathPrefix, $realPath);
+                return new ThemeFileInfo($realPath, $virtualPath, $category);
+            }
+        }
+
+        // The path didn't belong to any category
+        return null;
+    }
+
+    /**
+     * Sanitizes the theme options making sure that paths are valid.
+     * Options that match a file will be cleared if the file is not valid,
+     * or replaced with the virtual path if the file is valid.
+     */
+    public function sanitizeImagePathsOnJson($attribute, $params)
+    {
+        $decodedOptions = json_decode($this->$attribute, true);
+        if (is_array($decodedOptions)) {
+            foreach($decodedOptions as &$value) {
+                if (empty($value) || $value == 'inherit') {
+                    continue;
+                }
+                $virtualPath = $this->getVirtualThemeFilePath($value);
+                if (!empty($virtualPath)) {
+                    $value = $virtualPath;
+                } elseif ($this->isVirtualPath($value) || realpath($value) !== false || realpath(Yii::app()->getConfig('rootdir') . '/' . $value) !== false) {
```

Comment:
// Alternative B - If any of the following:
- option value matches a virtual path format or 
- option value matches a real existing path to a file either relative to the root LS installation or to the current workgin dir or absolute

Mark the value as invalid, as that's not allowed